lib/http2/connection.c

```diff
@@ -859,6 +866,7 @@ static void on_read(h2o_socket_t *sock, const char *err)
     h2o_http2_conn_t *conn = sock->data;
 
     if (err != NULL) {
+        h2o_context_report_http2_error(conn->super.ctx, H2O_HTTP2_ERROR_OTHER);
```

Comment:
Do we need to log read / write errors?

If so, I would suggest separating the structures that count the HTTP/2 error codes and the internal errors (read / write errors), rather than adding a privately-defined slot to the list HTTP/2 error codes.

Read / write errors could be logged more directly (for example by incrementing the counters directly like `++ctx->http2.events.read_closed`).  And also, the code and the structure for counting the HTTP/2 protocol-level errors could be renamed to better reflect the purpose.